commit 4e9cc64d8d6edebf03c408bc42d12d487d18e496 (refs/changes/09/3209/4)
Author: Orlando Vazquez <ovazquez@gmail.com>
Date:   2018-02-02T11:27:10-08:00 (1 year, 8 months ago)
    
    CNAPI-733 Add error for when cn-agent indicates docker cp'ing is not supported on a stopped container
    Reviewed by: Josh Wilsdon <josh@wilsdon.ca>

diff --git a/lib/endpoints/vms.js b/lib/endpoints/vms.js
index 3ba821e..414e651 100644
--- a/lib/endpoints/vms.js
+++ b/lib/endpoints/vms.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2017, Joyent, Inc.
+ * Copyright (c) 2018, Joyent, Inc.
  */
 
 /*
@@ -1053,6 +1053,14 @@ VM.dockerCopy = function handlerVmDockerCopy(req, res, next) {
                 return;
             }
 
+            if (details.restCode ===
+                    'DockerCopyStoppedContainerNoPlatformSupport')
+            {
+                next(new
+                    errors.DockerCopyStoppedContainerNoPlatformSupportError());
+                return;
+            }
+
             next(new restify.InternalError(error.message));
             return;
         }
diff --git a/lib/errors.js b/lib/errors.js
index 0b83716..79efec4 100644
--- a/lib/errors.js
+++ b/lib/errors.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2015, Joyent, Inc.
+ * Copyright (c) 2018, Joyent, Inc.
  */
 
 var assert = require('assert-plus');
@@ -210,6 +210,26 @@ FileNotFoundError.restCode = 'FileNotFound';
 FileNotFoundError.statusCode = 404;
 FileNotFoundError.description = 'could not find requested file';
 
+var unsupportedDockerCopyStoppedMessage
+    = 'This container must be running in order to use `docker cp`';
+function DockerCopyStoppedContainerNoPlatformSupportError() {
+    _CnapiBaseError.call(this, {
+        restCode: this.constructor.restCode,
+        statusCode: this.constructor.statusCode,
+        message: unsupportedDockerCopyStoppedMessage
+    });
+}
+
+util.inherits(
+    DockerCopyStoppedContainerNoPlatformSupportError, _CnapiBaseError);
+DockerCopyStoppedContainerNoPlatformSupportError.prototype.name =
+    'DockerCopyStoppedContainerNoPlatformSupportError';
+DockerCopyStoppedContainerNoPlatformSupportError.restCode =
+    'DockerCopyStoppedContainerNoPlatformSupport';
+DockerCopyStoppedContainerNoPlatformSupportError.statusCode = 409;
+DockerCopyStoppedContainerNoPlatformSupportError.description =
+    unsupportedDockerCopyStoppedMessage;
+
 
 function PathNotDirectoryError() {
     _CnapiBaseError.call(this, {
@@ -244,5 +264,7 @@ module.exports = {
     VmNotRunningError: VmNotRunningError,
     ServerNotRunningError: ServerNotRunningError,
     FileNotFoundError: FileNotFoundError,
-    PathNotDirectoryError: PathNotDirectoryError
+    PathNotDirectoryError: PathNotDirectoryError,
+    DockerCopyStoppedContainerNoPlatformSupportError:
+        DockerCopyStoppedContainerNoPlatformSupportError
 };
diff --git a/package.json b/package.json
index ebfc4dd..deadcfe 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "cnapi",
   "description": "SmartDataCenter Compute Node API",
-  "version": "1.5.1",
+  "version": "1.5.2",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
